analyzers/src/SonarAnalyzer.Common/Rules/DeclareTypesInNamespacesBase.cs

```diff
@@ -43,8 +42,7 @@ public abstract class DeclareTypesInNamespacesBase<TSyntaxKind> : SonarDiagnosti
               c =>
               {
                   var declaration = c.Node;
-
-                  if (c.ContainingSymbol.Kind != SymbolKind.NamedType
+                  if (c.IsRedundantPositionalRecordContext()
```

Comment:
`SyntaxNodeAnalysisContextExtensions.cs` needed to be moved from `SonarAnalyzer.CSharp` to `SonarAnalyzer.Common` to make `IsRedundantPositionalRecordContext` accessible here.